test/framework/src/main/java/org/elasticsearch/transport/nio/channel/ChannelFactory.java

```diff
@@ -114,7 +115,12 @@ private void scheduleServerChannel(NioServerSocketChannel channel, AcceptingSele
         SocketChannel openNioChannel(InetSocketAddress remoteAddress) throws IOException {
             SocketChannel socketChannel = SocketChannel.open();
             configureSocketChannel(socketChannel);
-            PrivilegedSocketAccess.connect(socketChannel, remoteAddress);
+            try {
```

Comment:
should we move the `configureSocketChannel` call into the try block as well here? seems cleaner